### DISCUSSION TITLE
[SUPPORT]: Custom fonts not working

### DISCUSSION THREAD

**Discussion by royscheeren** [dompdf/dompdf#3560] [Q&A] [answered]
Hi,

Is there somebody able to tell me why I can't get custom fonts to work in this repo:
https://github.com/royscheeren/pdf-font-test

Its based on this example:
https://www.itsolutionstuff.com/post/laravel-dompdf-add-custom-font-family-exampleexample.html

I feel like I've tried everything suggested that I've been able to find.

Thank you

**Comment by bsweeney** [comment]
I don't use laravel, so I can only get you so far.

The template you have specified the path to the file as `fonts/Croissant_One/CroissantOne-Regular.ttf` meaning it would be relative to the current URL path. Are files in storage by default accessible via HTTP? If not did you execute the appropriate command to set up a link to those files?

Did you try just outputting the HTML so you could see if it renders as expected in the browser? You can do that by replacing `return $pdf->stream();` with `return $pdf->outputHtml();`. Doing this I see that the HTML doesn't render with that font either, which leads me to believe that the font path you used in your template isn't valid. Without mapping paths or anything, I think what you actually want to do with a quick-start project like this is put the font files in the "public" directory not the "storage" directory. "storage/fonts" is used by Dompdf to store generated files.

**Reply by royscheeren** [reply]
Thanks for your reply @bsweeney. It took a bit of time before I could test again, but I've updated the above repo with a completely simplified example (no Laravel) and I still seem to miss something.

**Reply by bsweeney** [reply, marked as answer]
In pdf-remote.php you're using `loadHtml` to load the document. This is similar to loading the file from the current directory of the filesystem. The font references use relative paths so those will be loaded via filesystem. In order to load those files the chroot option must include the path to the file (e.g., `'chroot' => realpath('.')` should work for your example).

There is some information in the wiki related to resource access: https://github.com/dompdf/dompdf/wiki/Usage#resource-references-and-uri-validation

**Reply by royscheeren** [reply]
Thank you, looks like this was my missing piece.
